pytest_rpc/helpers.py

```diff
@@ -216,10 +217,180 @@ def delete_instance(instance_name, run_on_host):
                                       action on.
 
     Raises:
-        AssertionError: If operation unsucessful.
+        AssertionError: If operation unsuccessful.
     """
     instance_id = get_id_by_name('server', instance_name, run_on_host)
     cmd = "{} openstack server delete {}'".format(utility_container,
                                                   instance_id)
+    run_on_host.run_expect([0], cmd)
+
+    assert (verify_if_deleted('server', instance_name, run_on_host))
+
+
+def verify_if_deleted(service_type, service_name, run_on_host, retries=10):
```

Comment:
The cause of the instance not being present in the list is not necessarily the result of a delete event. Can this logic be combined with the `verify_asset_in_list` method also in this PR?

Reply:
@johnduarte : Updated!